pkg/trace/api/otlp.go

```diff
@@ -0,0 +1,494 @@
+	if cfg == nil {
+		cfg = new(config.OTLP)
+	}
+	return &OTLPReceiver{out: out, cfg: cfg}
+}
+
+// Start starts the OTLPReceiver, if any of the servers were configured as active.
+func (o *OTLPReceiver) Start() {
+	if o.cfg.HTTPPort != 0 {
+		o.httpsrv = &http.Server{
+			Addr:    fmt.Sprintf("%s:%d", o.cfg.BindHost, o.cfg.HTTPPort),
+			Handler: o,
+		}
+		o.wg.Add(1)
+		go func() {
+			defer o.wg.Done()
+			if err := o.httpsrv.ListenAndServe(); err != nil {
+				if err != http.ErrServerClosed {
+					log.Criticalf("Error starting OpenTelemetry HTTP server: %v", err)
+				}
+			}
+		}()
+		log.Infof("OpenTelemetry HTTP receiver running on http://%s:%d", o.cfg.BindHost, o.cfg.HTTPPort)
+	}
+	if o.cfg.GRPCPort != 0 {
+		ln, err := net.Listen("tcp", fmt.Sprintf("%s:%d", o.cfg.BindHost, o.cfg.GRPCPort))
+		if err != nil {
+			log.Criticalf("Error starting OpenTelemetry gRPC server: %v", err)
+		} else {
+			o.grpcsrv = grpc.NewServer()
+			otlppb.RegisterTraceServiceServer(o.grpcsrv, o)
+			o.wg.Add(1)
+			go func() {
+				defer o.wg.Done()
+				if err := o.grpcsrv.Serve(ln); err != nil {
+					log.Criticalf("Error starting OpenTelemetry gRPC server: %v", err)
+				}
+			}()
+			log.Infof("OpenTelemetry gRPC receiver running on %s:%d", o.cfg.BindHost, o.cfg.GRPCPort)
+		}
+	}
+}
+
+// Stop stops any running server.
+func (o *OTLPReceiver) Stop() {
+	if o.httpsrv != nil {
+		timeout, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		go func() {
+			o.httpsrv.Shutdown(timeout)
+			cancel()
+		}()
+	}
+	if o.grpcsrv != nil {
+		go o.grpcsrv.Stop()
+	}
+	o.wg.Wait()
+}
+
+// Export implements otlppb.TraceServiceServer
+func (o *OTLPReceiver) Export(ctx context.Context, in *otlppb.ExportTraceServiceRequest) (*otlppb.ExportTraceServiceResponse, error) {
+	defer timing.Since("datadog.trace_agent.otlp.process_grpc_request_ms", time.Now())
+	md, _ := metadata.FromIncomingContext(ctx)
+	metrics.Count("datadog.trace_agent.otlp.payload", 1, tagsFromHeaders(http.Header(md), otlpProtocolGRPC), 1)
+	o.processRequest(otlpProtocolGRPC, http.Header(md), in)
+	return &otlppb.ExportTraceServiceResponse{}, nil
+}
+
+// ServeHTTP implements http.Handler
+func (o *OTLPReceiver) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	defer timing.Since("datadog.trace_agent.otlp.process_http_request_ms", time.Now())
+	mtags := tagsFromHeaders(req.Header, otlpProtocolHTTP)
+	metrics.Count("datadog.trace_agent.otlp.payload", 1, mtags, 1)
+
+	rd := NewLimitedReader(req.Body, o.cfg.MaxRequestBytes)
+	slurp, err := ioutil.ReadAll(rd)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		metrics.Count("datadog.trace_agent.otlp.error", 1, append(mtags, "reason:read_body"), 1)
+		return
+	}
+	metrics.Count("datadog.trace_agent.otlp.bytes", int64(len(slurp)), mtags, 1)
+	var in otlppb.ExportTraceServiceRequest
+	switch getMediaType(req) {
+	case "application/x-protobuf":
```

Comment:
I just want to confirm that we are supporting here , per the compliance matrix: (here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md#exporters) 

`OTLP/HTTP binary Protobuf Exporter` format. Is that correct?